### PR TITLE
fix(osc): enhance osc feature

### PR DESCRIPTION
**Pull request by LioRoger**
osc support lock table for ob-mysql 4.2.5, fix data loss issue when switch table, fix thread leak issue when release Oms project

#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
1. support lock table for ob-mysql 4.2.5
2. fix data loss issue when switch table,
3. fix thread leak issue when release OMS project
4. support special schema/table

#### Which issue(s) this PR fixes:

Fixes #
1. support lock table for ob-mysql with version greater than 4.2.5
2. cause the missing check of data apply progress，data may loss during table switch,. Now we will wait until increment checkpoint in timestamp greater than the timestamp recorded after lock table. Notice, this waiting may take about 10 - 20 seconds now. We will optimize this lock table waiting time in future release.
3. fix thread leak when release OMS project. change to sync call instead of create a new Thread pool.
4. add escape character to MySQL/Oracle schema/table name

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by LuckyPickleZZ — DISMISSED** [review]
Excellent unit test samples!

**Review by yizhouxw — COMMENTED** [review]
nice improvement!

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/rename/SwapTableUtil.java [hunk 1]
```diff
@@ -35,4 +37,31 @@ public static boolean isSwapTableEnable(SwapTableType swapTableType, TaskStatus
                 (fullVerificationResult == FullVerificationResult.CONSISTENT ||
                         fullVerificationResult == FullVerificationResult.UNCHECK);
     }
+
+    public static String escapeMySQLName(String name) {
```
[the remark this answers is not in the document]
Comment (LioRoger) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/rename/SwapTableUtil.java]: that's a typo. getQuota has been removed. use StringUtils class wrap to reduce duplicate code

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by LuckyPickleZZ — APPROVED** [review]

**Pull request merged**